iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BatteryImpl.java

```diff
@@ -76,19 +80,20 @@ public Battery setP0(double p0) {
      */
     @Override
     public double getQ0() {
-        return q0.get(getNetwork().getVariantIndex());
+        return q0.get(network.getVariantIndex());
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public Battery setQ0(double q0) {
-        ValidationUtil.checkQ0(this, q0);
-        int variantIndex = getNetwork().getVariantIndex();
+        ValidationUtil.checkQ0(this, q0, network.getMinValidationLevel().compareTo(ValidationLevel.LOADFLOW) >= 0);
+        int variantIndex = network.getVariantIndex();
         double oldValue = this.q0.set(variantIndex, q0);
-        String variantId = getNetwork().getVariantManager().getVariantId(variantIndex);
+        String variantId = network.getVariantManager().getVariantId(variantIndex);
         notifyUpdate("q0", variantId, oldValue, q0);
+        network.invalidate();
```

Comment:
invalidate is confusing. First time I looked at the code it was not clear what kind of invalidation it was (connected components, a topo cache?)
`invalidateValidationLevel()` ?